torch/csrc/distributed/c10d/CUDASymmetricMemory-inl.h

```diff
@@ -33,6 +36,9 @@ cas(uint32_t* addr, uint32_t compare, uint32_t val) {
   cuda::atomic_ref<uint32_t, cuda::thread_scope_system> ref(*addr);
   ref.compare_exchange_strong(compare, val, cuda::std::memory_order(Sem));
   return compare;
+#elif defined(USE_ROCM)
+  __atomic_compare_exchange_n(addr, &compare, val, false, Sem, Sem);
```

Comment:
I'm not sure why this compiles (probably related to the C++ version, but __atomic_compare_exchange_n takes int as memory order while Sem is std::memory_order. I tried convert to int, but the fail_memoryorder cannot be release - so I hardcoded it to relaxed